src/Parallel/PhaseControl/PhaseControlTags.hpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <type_traits>
+#include <utility>
+
+#include "Parallel/Reduction.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace PhaseControl {
+
+/// A type for denoting a piece of data for deciding a phase change.
+///
+/// `Tag` is intended to be a tag (with a type) for indexing a
+/// `tuples::TaggedTuple`, and `CombineMethod` is intended to be a
+/// `Parallel::ReductionDatum`-compatible invokable for combining the `type` of
+/// the `Tag`. The `MainCombineMethod` is provided to give flexibility for a
+/// different method of combination at the top level of the hierarchy (so, in
+/// the case of phase control reductions, performed by the main chare to combine
+/// reductions from different chares)
+template <typename Tag, typename CombineMethod,
+          typename MainCombineMethod = CombineMethod>
+struct TagAndCombine {
+  using tag = Tag;
+  using combine_method = CombineMethod;
+  using main_combine_method = MainCombineMethod;
+};
+
+namespace detail {
+template <typename T>
+struct get_tag {
+  using type = typename T::tag;
+};
+}  // namespace detail
+
+/// Extracts a list of tags from a list of `PhaseControl::TagAndCombine`s
+template <typename TagsAndCombines>
+using get_tags = tmpl::transform<TagsAndCombines, detail::get_tag<tmpl::_1>>;
+
+// @{
+/// A flexible combine invokable that combines into a `tuples::TaggedTuple` a
+/// new `tuples::TaggedTuple` with a subset of the original tags, and combines
+/// according to a list of `PhaseControl::TagAndCombine`s.
+template <typename TagsAndCombines>
+struct TaggedTupleCombine {
+  template <typename... CurrentTags, typename... CombineTags>
+  tuples::TaggedTuple<CurrentTags...> operator()(
+      tuples::TaggedTuple<CurrentTags...> current_state,
+      const tuples::TaggedTuple<CombineTags...>& element) {
+    tmpl::for_each<tmpl::list<CombineTags...>>(
+        [&current_state, &element](auto tag_v) noexcept {
+          using tag = typename decltype(tag_v)::type;
+          tuples::get<tag>(current_state) = typename tmpl::at<
```

Comment:
from a call, conclusion was put combine_method (and possibly main_combine_method) into the Tags